### PR TITLE
chore/misc: adds missing changelog item, adds missing return statemen…

### PR DESCRIPTION
**Pull request by hunterlester**
…ts to tests

### CODE REVIEW

File: test/mutable/entries_keys_values.js [hunk 1]
```diff
@@ -97,15 +94,15 @@ describe('Mutable Data Entries', () => {
...
+  it('check list of values', () => {
     app.mutableData.newRandomPublic(TYPE_TAG)
       .then((m) => m.quickSetup(TEST_ENTRIES).then(() => m.getValues()))
       .then((values) => Promise.all(values.map((value) =>
         should(TEST_ENTRIES).matchAny((v) => {
           should(v).be.eql(value.buf.toString());
-          should(value.version).be.equal(0);
+          return should(value.version).be.equal(0);
         })
-      )).then(() => done(), (err) => done(err)));
```
Comment (bochaco) [inline test/mutable/entries_keys_values.js]: Be careful here, this will give you false positives.

File: test/mutable/permissions.js [hunk 2]
```diff
@@ -33,14 +33,13 @@ describe('Permissions', () => {
...
 
-  it('list permission sets and remove them', (done) => {
+  it('list permission sets and remove them', () => {
     app.mutableData.newRandomPublic(TYPE_TAG)
       .then((m) => m.quickSetup(TEST_ENTRIES).then(() => m.getPermissions())
         .then((perms) => perms.listPermissionSets()
           .then((permSets) => Promise.all(permSets.map((userPermSet) =>
             should(m.delUserPermissions(userPermSet.signKey, 1)).be.fulfilled()
           )))
-          .then(() => done(), (err) => done(err))
```
Comment (bochaco) [inline test/mutable/permissions.js]: This will give you false positives as well

File: test/mutable/entries_keys_values.js [hunk 3]
```diff
@@ -79,13 +79,10 @@ describe('Mutable Data Entries', () => {
       .then((keys) => should(keys.length).equal(Object.keys(TEST_ENTRIES).length))
   );
 
-  it('check list of keys', (done) => {
-    app.mutableData.newRandomPublic(TYPE_TAG)
+  it('check list of keys', () => app.mutableData.newRandomPublic(TYPE_TAG)
       .then((m) => m.quickSetup(TEST_ENTRIES).then(() => m.getKeys()))
       .then((keys) => Promise.all(keys.map((key) =>
-        should(TEST_ENTRIES).have.ownProperty(key.toString())
-      )).then(() => done(), (err) => done(err)));
```
Comment (bochaco) [inline test/mutable/entries_keys_values.js]: This will give you false positives

Comment (hunterlester) [reply test/mutable/entries_keys_values.js]: I'm able to fail this test if I give an incorrect value to `should(TEST_ENTRIES).have.ownProperty(...`

**Pull request merged**